viewing_party/party.py

```diff
@@ -1,23 +1,122 @@
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
+def create_movie(movie_title, genre, rating):
+    if not movie_title or not genre or not rating:
+        return None 
+    new_movie = {
+        "title" : movie_title,
+        "genre" : genre,
+        "rating" : rating}
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
 
-def create_movie(title, genre, rating):
-    pass
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
+def watch_movie(user_data, title):
+    for movie_to_watch in user_data["watchlist"]:
+        if movie_to_watch["title"] == title:
+            user_data["watchlist"].remove(movie_to_watch)
+            user_data["watched"].append(movie_to_watch)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = 0
+    if not user_data["watched"]:
+        return 0
+
+    watched_movie = user_data["watched"]
+    for movie in watched_movie:
+        rating += movie["rating"]
+
+    movie_len = len(user_data["watched"])
+    avg_rating = rating / movie_len
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    count = {}

```

Comment:
`count` can be a reserved keyword in other languages. In Python3, count() is a list method. To avoid potential confusion from our teammates or even the tools we use, we should generally avoid using names that could be a reserved keyword or an existing method.